specification.md

```diff
@@ -1859,22 +1860,33 @@ The `version` property can be used to provide a specific workflow version. It mu
 The `annotations` property defines a list of helpful terms describing the workflows intended purpose, subject areas, or other important qualities,
 for example "machine learning", "monitoring", "networking", etc
 
-The `dataInputSchema` property can be used to validate the workflow data input against a defined JSON Schema.
-This check should be done before any states are executed. `dataInputSchema` can have two different types.
-If it is an object type it has the following definition:
+The `inputDataSchema` and `outputDataSchema` properties can be used to validate input and output data against a defined JSON Schema.
+
+The `inputDataSchema` property validates the [workflow data input](#Workflow-Data-Input). Validation should be performed before any states are executed. In case of
+a start [Event state](#Event-state) the input schema is ignored, if present. The `failOnValidationErrors` property  determines if workflow execution should continue in case of validation errors. 
+
+The `outputDataSchema` property validates the [Workflow data output](#workflow-data-output). Validation should be performed after workflow execution has finished successfully. Successfully means that the workflow has completed an end state without errors. 
+The `failOnValidationErrors` property determines what should be done when the workflow output does not match the provided schema. 
+If `failOnValidationErrors` is true, an error should be thrown. If executed within a subprocess, that error might be handled by the parent workflow. 
```

Comment:
"thrown" is kinda java-ish. pls use "raised"

Reply:
Totally disagree here! Errors are not raised, they are thrown, it's not java-ish, it's IT-ish and is the proper terminology in that case imo

Reply:
As we are speaking of error control flow, the terminology `catch` is actually more appropriate. 
As a matter of fact, raising an error is the right terminology when "producing" the error.

Btw, `throw`, `catch` and other keywords of the like are NOT specific to Java or a specific grammar. Those are specific IT terms, very widely and globally used, across languages and techs, to describe, in this case, error control flow.  It is not a matter of opinion, it's a matter of proper vocabulary. Making proper use of that vocabulary ensures ubiquitousness. Using synonyms for the sake of not looking like code is  IMHO very dangerous, misleading to the biggest part of our community (devs mostly) and is even confusing.

Reply:
there is no throw and catch in Go :)

Reply:
@tsurdilo indeed, because there is no proper error handling out of the box. 
 https://link.medium.com/nkJbfoMHuvb

Doesn't mean that because they do not provide a core feature that said feature is wrongly named 😜

Reply:
| They are basic IT terms

yes exactly, thats why we should if possible stay away from using them inside spec text. DSLs are to be read and understood by non-IT people too

Reply:
But dude, catch and throw are less misleading that synonyms that do not apply to the concepts.
Again, it's not because something is rightfully used by global consensus in IT that we should not use it and confuse most ppl. 
Those terms apply perfectly, are easily understood by even the dumbest guy around, and avoids guaranteed misinterpretation.

Reply:
It would be also easier to use "return error" and "check error" and "panic" and "unwrap" too but i guess this maybe too much. So idk you seem to have a strong opinion about this so not going to argue with you, but don't think try/catch is really that "mainstream" outside devel world maybe

Reply:
Well, I do have a very strong opinion on that matter, but that's just me, and I'm indeed a bit of a grammar nazy 😜 
Let's see what the others think about it, and let's settle this out in a weekly. 
Wdyt?

Reply:
Suggested terms are all far fetched imho, and do not really paint the 'action' we want to describe, again imho.

Reply:
Sure, lets try that and catch any issues that come up ;)

Reply:
Jokes appart, I hear you and I do believe you make a very valid point by always thinking of not turning the spec into some kind of IT only language. 
I just think that in this specific case, as the grammar nazy I must admit I am, alternatives sounds a bit far fetched whereas the try/catch/throw one is beautifully simple and ubiquitous. 
Let's see what the others think and settle that out in a weekly, wdyt?

Reply:
I can use "generated" rather than "thrown", is that ok for you guys?

Reply:
I don't think that's right either. I like the thrown term. But whathever, it's just a minor semantic detail, so I'll go with whatever you guys settle on 😉

Reply:
I do not have particular interest on discussing a semantic thing, so Tiho let us know which is your chosen word. 
If it helps, thow was not invented by Java, it was chosen for C++. And Java also includes (inherited from other languages) english words like if, else, finally, int, float, double, string, array... ;)

Reply:
I'd rather stay with the ubiquitous term, which are try/catch/throw, widely used by the IT community. Raised is yet another good term, thou. But in this context, throw is the best one. And just to try to add a period in this conversation: https://english.stackexchange.com/questions/319126/why-exactly-is-the-verb-to-throw-used-when-an-error-occurs-in-the-program.